lib/rules/no-duplicate-class-names.js

```diff
@@ -0,0 +1,319 @@
+      }
+      break
+    }
+    case 'BinaryExpression': {
+      if (nodeExpression.operator === '+') {
+        yield* extractClassNodes(node, nodeExpression.left, unconditional)
+        yield* extractClassNodes(node, nodeExpression.right, unconditional)
+      }
+      break
+    }
+  }
+}
+
+/**
+ * @param {string} classList
+ * @returns {string[]}
+ */
+function getClassNames(classList) {
+  return classList.split(/\s+/).filter(Boolean)
+}
+
+/**
+ * @param {string} raw - raw class names string including quotes
+ * @returns {string}
+ */
+function removeDuplicateClassNames(raw) {
+  const quote = raw[0]
+  const inner = raw.slice(1, -1)
+  const tokens = inner.split(/(\s+)/)
+
+  /** @type {string[]} */
+  const kept = []
+  const used = new Set()
+
+  for (let i = 0; i < tokens.length; i++) {
+    const token = tokens[i]
+    if (!token) continue
+
+    const isWhitespace = /^\s+$/.test(token)
+
+    if (isWhitespace) {
+      // add whitespace to the last kept item or as leading whitespace
+      if (kept.length > 0) {
+        kept[kept.length - 1] += token
+      } else {
+        kept.push(token)
+      }
+    } else if (used.has(token)) {
+      // handle duplicate class name
+      const nextToken = tokens[i + 1]
+      const hasNextWhitespace =
+        kept.length > 0 && i + 1 < tokens.length && /^\s+$/.test(nextToken)
+
+      if (hasNextWhitespace) {
+        // update spaces of the last non-whitespace item
+        for (let j = kept.length - 1; j >= 0; j--) {
+          const isNotWhitespace = !/^\s+$/.test(kept[j])
+          if (isNotWhitespace) {
+            const parts = kept[j].split(/(\s+)/)
+            kept[j] = parts[0] + nextToken
+            break
+          }
+        }
+        i++ // skip the whitespace token
+      }
+    } else {
+      kept.push(token)
+      used.add(token)
+    }
+  }
+
+  // remove trailing whitespace from the last item if it's not purely whitespace
+  // unless the original string ended with whitespace
+  const endsWithSpace = /\s$/.test(inner)
+  if (kept.length > 0 && !endsWithSpace) {
+    const lastItem = kept[kept.length - 1]
+    const isLastWhitespace = /^\s+$/.test(lastItem)
+    if (!isLastWhitespace) {
+      const parts = lastItem.split(/(\s+)/)
+      kept[kept.length - 1] = parts[0]
+    }
+  }
+
+  return quote + kept.join('') + quote
+}
+
+/** @param {VLiteral | Literal | TemplateElement | null} node */
+function getRawValue(node) {
+  if (!node?.value) return null
+  return typeof node.value === 'object' && 'raw' in node.value
+    ? node.value.raw
+    : node.value
+}
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      url: 'https://eslint.vuejs.org/rules/no-duplicate-class-names.html',
+      description: 'disallow duplication of class names in class attributes',
+      categories: undefined
+    },
+    fixable: 'code',
+    schema: [],
+    messages: {
+      duplicateClassNames: 'Duplicate class names {{names}}.'
```

Comment:
Sorry I wasn't clear enough: The error message should be `Duplicate class names 'foo', 'bar'.` but `Duplicate class name 'foo'.`.